web3/module.py

```diff
@@ -1,11 +1,32 @@
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Coroutine,
+    Dict,
+    Union,
+)
+
 from eth_utils.toolz import (
     curry,
     pipe,
 )
 
+from web3.method import (
+    Method,
+)
+from web3.types import (
+    JsonRpcResponse,
+)
+
+if TYPE_CHECKING:
+    from web3 import Web3  # noqa: F401
+
 
 @curry
-def apply_result_formatters(result_formatters, result):
+def apply_result_formatters(
+    result_formatters: Dict[str, Any], result: JsonRpcResponse
```

Comment:
The type for `result_formatters` looks wrong here.  `pipe` expects a value as the first argument and then all the rest are expected to be callables...